ESXi ART Tests Batch 1

blueteam0ps opened the pull request:
**Details:**
Adding the following ART tests which are to be run against VMWARE ESXi. Note that the tests were built to be run on Windows and leverages PowerCLI and also Plink/PSCP to communicate with the remote ESXi host.

ESXi - Enable SSH via PowerCLI
ESXi - VM Discovery using ESXCLI
ESXi - Darkside system information discovery
ESXi - Enumerate VMDKs available on an ESXi Host
ESXi - Install a custom VIB on an ESXi host 
ESXi - Terminates VMs using pkill
ESXi - Avoslocker enumerates VMs and forcefully kills VMs
ESXi - Disable Account Lockout Policy via PowerCLI
ESXi - Change VIB acceptance level to CommunitySupported via PowerCLI
ESXi - Change VIB acceptance level to CommunitySupported via ESXCLI

**Testing:**
Testing performed locally and screenshot of test execution are attached to the PR.
![T1021 004-1](https://github.com/redcanaryco/atomic-red-team/assets/1480956/e0407aab-4b26-4bf5-995a-86bb505aaf05)
![T1082-32-discovery](https://github.com/redcanaryco/atomic-red-team/assets/1480956/336e2dd8-9341-402f-b68f-331b79030ce0)
![T1082-33-darksidediscovery](https://github.com/redcanaryco/atomic-red-team/assets/1480956/95da9f8f-e884-4455-814d-51d3604abca1)
![T1083-find](https://github.com/redcanaryco/atomic-red-team/assets/1480956/bc5f9e31-3643-4c27-b18e-ecd2d4064e76)
![T1129](https://github.com/redcanaryco/atomic-red-team/assets/1480956/9acfebcc-b32f-4b15-a4e6-77d0cebf6512)
![T1529-13](https://github.com/redcanaryco/atomic-red-team/assets/1480956/d00d44a5-f969-41c7-976b-002000da2ad9)
![T1529-14](https://github.com/redcanaryco/atomic-red-team/assets/1480956/e5b8e4b8-3669-415d-ba1d-33bf80fca098)
![T1562 001-50](https://github.com/redcanaryco/atomic-red-team/assets/1480956/08d1cc0c-f501-4028-8c5a-867a8b3157d9)
![T1562 010-1](https://github.com/redcanaryco/atomic-red-team/assets/1480956/bdbb9b71-3902-42a8-92a0-960ff108ced8)
![T1562 010-2](https://github.com/redcanaryco/atomic-red-team/assets/1480956/a1e0aa1c-77a9-4d4c-83c6-faf5e8f9f497)


**Associated Issues:**
Only errors you might get in the discovery tests as your environment may not have SAN storage.

blueteam0ps committed:
ESXi Tests Batch 1

clr2of8 committed:
remove duplicate key

clr2of8 committed:
Merge branch 'master' into esxi1

clr2of8 reviewed (changes requested):
These look great. Just one small request ... Please have the prereqs download plink and pscp executables to the ExternalPayloads folder instead of including it in this repo in the various bin directories.

blueteam0ps committed:
Update T1082.yaml

Updated the binary location to ExternalPayloads folder and also added the folder creation in GetPreReqs

blueteam0ps committed:
Update T1083.yaml

Added External Payloads and included folder creation in GetPreReqs

blueteam0ps committed:
Update T1129.yaml

Added ExternalPayloads refence. 
Added folder creation in GetPreReqs
Move the reference of the vib to src

blueteam0ps committed:
Update T1529.yaml

Added External Payloads folder and added folder creation step to GetPreReqs

blueteam0ps committed:
Update T1529.yaml

987c9b4d-a637-42db-b1cb-e9e242c3991b - added external payloads

blueteam0ps committed:
Update T1562.010.yaml

Added External Payloads reference and folder creation to GetPreReqs

blueteam0ps committed:
Moved the vib to src

blueteam0ps committed:
Delete atomics/T1129/bin directory

blueteam0ps committed:
Delete atomics/T1082/bin directory

blueteam0ps committed:
Delete atomics/T1083/bin directory

blueteam0ps committed:
Delete atomics/T1562.010/bin directory

blueteam0ps reviewed (commented):
All good

blueteam0ps commented:
> These look great. Just one small request ... Please have the prereqs download plink and pscp executables to the ExternalPayloads folder instead of including it in this repo in the various bin directories.

Done and I've removed the bin folders too. Over to you ;)

clr2of8 committed:
Merge branch 'master' into esxi1
